core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -351,8 +351,7 @@ private void startNewPage()
     {
         if (currentPageBuilder != null) {
             completedPagesMemorySize += currentPageBuilder.getRetainedSizeInBytes();
-            // TODO: (https://github.com/trinodb/trino/issues/12484) pre-size new PageBuilder to OUTPUT_PAGE_SIZE
-            currentPageBuilder = currentPageBuilder.newPageBuilderLike();
+            currentPageBuilder.resetExact();
```

Comment:
use `currentPageBuilder.reset(PositionsAppenderUtils.calculateNewArraySize(declaredPositions))`

Reply:
So that we don't need `resetExact();` method, see https://github.com/trinodb/trino/pull/12512/commits/270656182e300e261f052ef6e54b1dd0d70fbec8#r922199657

Reply:
I meant `currentPageBuilder.reset(currentPageBuilder.getPositionCount())`